### PR TITLE
Add support for network zones

### PR DESCRIPTION
**Pull request by mandrav**
This patch add support for lxd_network_zone and lxd_network_zone_record resources.

### CODE REVIEW

**Comment by MusicDin** [comment]
I've quickly went through the code, and we should check whether the target LXD server supports the appropriate extension.

Something like this:
```go
nzExt := "projects_networks_zones"
if !server.HasExtension(nzExt) {
	return fmt.Errorf("Network zones cannot be used because LXD server does not support %q extension.", nzExt)
}
```

However, I'm not sure `projects_networks_zones` is the appropriate one.
@simondeziel could you help here?

File: docs/resources/resource_lxd_network_zone.go
Comment (adamcstephens) [inline docs/resources/resource_lxd_network_zone.go:1]: This go file shouldn't be here.

**Review by adamcstephens — CHANGES_REQUESTED** [review]
I haven't fully reviewed this yet, but you're missing the documentation for the zone and tests for the record. Also note the CI warnings/errors.

**Comment by simondeziel** [comment]
> I've quickly went through the code, and we should check whether the target LXD server supports the appropriate extension.
> 
> Something like this:
> 
> ```go
> nzExt := "projects_networks_zones"
> if !server.HasExtension(nzExt) {
> 	return fmt.Errorf("Network zones cannot be used because LXD server does not support %q extension.", nzExt)
> }
> ```
> 
> However, I'm not sure `projects_networks_zones` is the appropriate one. @simondeziel could you help here?

tl;dr: you've got the right extension IMHO.

The DNS support depends on `network_dns_records` but when only that extension is present, the network zone is created in the same project as the network it is associated with. The `projects_networks_zones` extension decouples the 2 which this PR seems to be using so I think only checking for `projects_networks_zones` is good.

File: lxd/resource_lxd_network_zone.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+	}
+}
+
+func resourceLxdNetworkZoneCreate(d *schema.ResourceData, meta interface{}) error {
+	p := meta.(*lxdProvider)
+	server, err := p.GetInstanceServer(p.selectRemote(d))
+	if err != nil {
+		return err
+	}
+
```
Comment (MusicDin) [inline lxd/resource_lxd_network_zone.go:60]: Can you please just add a check if server supports `projects_networks_zones` extension:

```go
netZoneExt := "projects_networks_zones"
if !server.HasExtension(netZoneExt) {
	return fmt.Errorf("Network zones cannot be used because LXD server does not support %q extension", netZoneExt)
}
```

File: lxd/resource_lxd_network_zone.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+	return resourceLxdNetworkZoneRead(d, meta)
+}
+
+func resourceLxdNetworkZoneRead(d *schema.ResourceData, meta interface{}) error {
+	p := meta.(*lxdProvider)
+	server, err := p.GetInstanceServer(p.selectRemote(d))
+	if err != nil {
+		return err
+	}
+
```
Comment (MusicDin) [inline lxd/resource_lxd_network_zone.go:94]: Same comment as above. I think adding extension check to Read and Create should be sufficient.

**Review by MusicDin — CHANGES_REQUESTED** [review]

File: lxd/resource_lxd_network_zone.go (hunk 2 above)
Comment (mandrav) [reply lxd/resource_lxd_network_zone.go:103]: Done.

File: docs/resources/network_zone_record.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+  config = {
+    "dns.nameservers"  = "ns.custom.example.org"
+    "peers.ns.address" = "127.0.0.1"
+  }
+}
+
+resource "lxd_network_zone_record" "record" {
+  name = "ns"
+  zone = lxd.network_zone.zone.id
```
Comment (MusicDin) [inline docs/resources/network_zone_record.md:25]: Just a small typo here: `lxd.network_zone` -> `lxd_network_zone`

File: docs/resources/network_zone.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+  config = {
+    "dns.nameservers"  = "ns.custom.example.org"
+    "peers.ns.address" = "127.0.0.1"
+  }
+}
+
+resource "lxd_network_zone_record" "record" {
+  name = "ns"
+  zone = lxd.network_zone.zone.id
```
Comment (MusicDin) [inline docs/resources/network_zone.md:25]: Just a small typo here: `lxd.network_zone` -> `lxd_network_zone`

**Review by MusicDin — CHANGES_REQUESTED** [review]

**Review by MusicDin — APPROVED** [review]

**Review by adamcstephens — APPROVED** [review]
LGTM. Thanks for your contribution! :pizza:
